nautobot_chatops_ipfabric/worker.py

```diff
@@ -427,6 +432,99 @@ def end_to_end_path(
     return True
 
 
+@subcommand_of("ipfabric")
+def pathlookup(
+    dispatcher, src_ip, dst_ip, src_port, dst_port, protocol
+):  # pylint: disable=too-many-arguments, too-many-locals
+    """Path simulation diagram lookup between source and target IP address."""
+    snapshot_id = get_user_snapshot(dispatcher)
+    sub_cmd = "pathlookup"
+    supported_protocols = ["tcp", "udp", "icmp"]
+    protocols = [(protocol.upper(), protocol) for protocol in supported_protocols]
+
+    dialog_list = [
+        {
+            "type": "text",
+            "label": "Source IP",
+        },
+        {
+            "type": "text",
+            "label": "Destination IP",
+        },
+        {
+            "type": "text",
+            "label": "Source Port",
+            "default": "1000",
+        },
+        {
+            "type": "text",
+            "label": "Destination Port",
+            "default": "22",
+        },
+        {
+            "type": "select",
+            "label": "Protocol",
+            "choices": protocols,
+            "default": protocols[0],
+        },
+    ]
+
+    if not all([src_ip, dst_ip, src_port, dst_port, protocol]):
+        dispatcher.multi_input_dialog(f"{BASE_CMD}", f"{sub_cmd}", "Path Lookup", dialog_list)
+        return CommandStatusChoices.STATUS_SUCCEEDED
+
+    # verify IP address and protocol is valid
+    if not is_ip(src_ip) or not is_ip(dst_ip):
+        dispatcher.send_error("You've entered an invalid IP address")
+        return CommandStatusChoices.STATUS_FAILED
+    if protocol not in supported_protocols:
+        dispatcher.send_error(f"You've entered an unsupported protocol: {protocol}")
+        return CommandStatusChoices.STATUS_FAILED
+
+    dispatcher.send_blocks(
+        [
+            *dispatcher.command_response_header(
+                f"{BASE_CMD}",
+                f"{sub_cmd}",
+                [
+                    ("src_ip", src_ip),
+                    ("dst_ip", dst_ip),
+                    ("src_port", src_port),
+                    ("dst_port", dst_port),
+                    ("protocol", protocol),
+                ],
+                "Path Lookup",
+                ipfabric_logo(dispatcher),
+            ),
+            dispatcher.markdown_block(f"{ipfabric_api.host_url}/diagrams/pathlookup"),
+        ]
+    )
+
+    # only supported in IP Fabric OS version 4.0+
+    response = ipfabric_api.get_os_version()
+    os_version = float(response.get("version", "0.0").rpartition(".")[0])
+    if os_version >= 4:
```

Comment:
regarding version handling, I was thinking about another potential approach and I would like to get your feedback.
Your proposal is:
- if os_version <x -> command_1
- if os_version >= x -> command_2
what about:
- command and behind the scenes we check the os_version and decide which logic to use? (this works well if the input data is the same)

Reply:
I'm totally open to a better solution. Originally I had the logic in the `end-to-end-path` command. However the new v4 OS names the path simulation a different name with a different URL. Also, we could extend the new path lookup for JSON and SVG output if desired. I felt it would be adding complexity, for the short term. But perhaps I'm not understanding your proposal?

Reply:
my idea would be to expose the user with only one command: `the_very_best_e2e_path`
Inside this we would have a version `if statement` that would call the specific logic that you need for `end-to-end-path`  and `pathlookup` (data input if there are changes, and the actual output)
The goal is to make the user experience "unique", so the user don' have to understand there are two implementations, there is one with different behaviour

Reply:
Ideally we'll extend the `pathlookup` command with JSON capability using an additional argument for output format. Unfortunately the JSON result for the new version and old version are completely different :-( 
So the only thing that's similar is the input variables which, will change very soon anyway.